sqlitecluster/SQLiteNode.cpp

```diff
@@ -1242,583 +1222,606 @@ bool SQLiteNode::update() {
+                PINFO("Re-queueing command '" << message["ID"] << "' (" << command->request.methodLine << ") ("
+                      << command->id << ")");
+                _server.acceptCommand(move(command), false);
+                _escalatedCommandMap.erase(commandIt);
+            } else
+                SWARN("Received ESCALATE_ABORTED for unescalated command " << message["ID"] << ", ignoring.");
+        } else if (SIEquals(message.methodLine, "CRASH_COMMAND") || SIEquals(message.methodLine, "BROADCAST_COMMAND")) {
+            // Create a new Command and send to the server.
+            SData messageCopy = message;
+            PINFO("Received " << message.methodLine << " command, forwarding to server.");
+            _server.acceptCommand(make_unique<SQLiteCommand>(move(messageCopy)), true);
         } else {
-            SHMMM("Received ESCALATE_RESPONSE for unknown command ID '" << message["ID"] << "', ignoring. ");
-        }
-    } else if (SIEquals(message.methodLine, "ESCALATE_ABORTED")) {
-        // ESCALATE_RESPONSE: Sent when the leader aborts processing an escalated command. Re-submit to the new leader.
-        if (_state != FOLLOWING) {
-            STHROW("not following");
-        }
-        if (!message.isSet("ID")) {
-            STHROW("missing ID");
-        }
-        PINFO("Received ESCALATE_ABORTED for '" << message["ID"] << "' (" << message["Reason"] << ")");
-
-        // Look for that command
-        auto lock = _escalatedCommandMap.scopedLock();
-        auto commandIt = _escalatedCommandMap.find(message["ID"]);
-        if (commandIt != _escalatedCommandMap.end()) {
-            // Re-queue this
-            unique_ptr<SQLiteCommand>& command = commandIt->second;
-            PINFO("Re-queueing command '" << message["ID"] << "' (" << command->request.methodLine << ") ("
-                  << command->id << ")");
-            _server.acceptCommand(move(command), false);
-            _escalatedCommandMap.erase(commandIt);
-        } else
-            SWARN("Received ESCALATE_ABORTED for unescalated command " << message["ID"] << ", ignoring.");
-    } else if (SIEquals(message.methodLine, "CRASH_COMMAND") || SIEquals(message.methodLine, "BROADCAST_COMMAND")) {
-        // Create a new Command and send to the server.
-        SData messageCopy = message;
-        PINFO("Received " << message.methodLine << " command, forwarding to server.");
-        _server.acceptCommand(make_unique<SQLiteCommand>(move(messageCopy)), true);
-    } else {
-        STHROW("unrecognized message");
+            STHROW("unrecognized message");
+        }
+    } catch (const SException& e) {
+        PWARN("Error processing message '" << message.methodLine << "' (" << e.what() << "), reconnecting.");
+        SData reconnect("RECONNECT");
+        reconnect["Reason"] = e.what();
+        peer->sendMessage(reconnect.serialize());
+        peer->shutdownSocket();
     }
```

Comment:
Were we not catching this before or is this necessary because we're doing ping/pong handling directly in this method now?

Reply:
We essentially moved the `try/catch` block inside `_onMESSAGE` from here: https://github.com/Expensify/Bedrock/pull/1252/files#diff-2d1c5f24a4b7ebc3076630d8ee3a1155ad5e7789abe72cff977e615541c319d6L2788-R2664